beesdoo_shift_swap/models/subscribed_underpopulated_shift.py

```diff
@@ -185,116 +160,59 @@ def subscribe_shift(self):
             *the user hasn't done another exchange 2month before
         :return:
         """
-        # Get the wanted shift
-        subscribed_shift_rec = self.comfirmed_shift_id
-
-        subscribed_shift_rec.is_regular = True
-
-        # Get the user
-        subscribed_shift_rec.worker_id = self.worker_id
+        if not self.confirme_status :
+            # Get the wanted shift
+            subscribed_shift_rec = self.confirmed_shift_id
+            subscribed_shift_rec.is_regular = True
+            # Get the user
+            subscribed_shift_rec.write({
+                "worker_id": self.worker_id
+            })
 
-        # Subscribe done, change the status
-        self.exchange_status = 1
+            # Subscribe done, change the status
+            self.confirme_status = 1
 
-        #update status
-        self.update_status()
-        if not self.exchanged_shift_id.worker_id and not self.exchange_status:
-            return False
+            #update status
+            self.update_status()
+            if not self.exchanged_shift_id.worker_id and not self.confirme_status:
+                return False
+            return True
         return True
 
     @api.multi
     def button_subscribe_shift(self):
         for exchange in self:
-            if not exchange.comfirmed_shift_id:
+            if not exchange.confirmed_shift_id:
                 raise Warning('Shift not generated')
             if not exchange.subscribe_shift():
                 raise Warning('cannot unsubscribe')
         return True
 
 
     def display_underpopulated_shift(self,my_timeslot):
```

Comment:
May be it should no be named `display` if it does not display something. :)

Reply:
remove

Reply:
remove button

Reply:
display -> get